Sofa/GL/src/sofa/gl/DrawToolGL.cpp

```diff
@@ -144,19 +144,25 @@ void DrawToolGL::drawLines(const std::vector<Vector3> &points, float size, const
         return drawLines(points, size, RGBAColor::red());
     }
 
-    glLineWidth(size);
-    if (getLightEnabled()) disableLighting();
-    glBegin(GL_LINES);
+    // gather lines with same colors
+    std::map<RGBAColor, std::vector<Vector3> > colorPointsMap;
+    for (std::size_t i = 0; i < colors.size(); ++i)
```

Comment:
Here is what I suggest to replace your loop:
```cpp
for (std::size_t i = 0; i < colors.size(); ++i)
{
    auto& colorPoints = colorsPointsMap[colors[i]];
    colorPoints.push_back(points[2 * i]);
    colorPoints.push_back(points[2 * i + 1]);
}
```

With your version: 3 searches in the map
With my version: 1 search in the map (and more concise)

Reply:
Sure that `     auto& colorPoints = colorsPointsMap[colors[i]];`  wont generate an error ?
I mean at the beginning there is no colors[i] in the map, and there is no std::vector<> allocated/built but maybe I am wrong

Reply:
See https://en.cppreference.com/w/cpp/container/map/operator_at. This operator performs an insertion if the key is not in the map.

Reply:
Okay at() was behaving like I thought operator[] was 🤫